src/router/router.tsx

```diff
@@ -0,0 +1,24 @@
+import AboutUsPage from '@/pages/about-us/AboutUsPage';
+import DetailsPage from '@/pages/details/Details';
+import ErrorPage from '@/pages/error/ErrorPage';
+import HomePage from '@/pages/home/HomePage';
+
+const router = [
+  {
+    path: '/',
+    element: <HomePage />,
+    errorElement: <ErrorPage />,
+    children: [
+      {
+        path: ':cardId',
```

Comment:
To get 404 page if url wrong (for example '/temp') it is better to modify child url
```suggestion
        path: 'card/:cardId',
```